src/editors/containers/VideoUploadEditor/index.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDropzone } from 'react-dropzone';
+import { defineMessages } from 'react-intl';
+import { FormattedMessage, injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { Button } from '@edx/paragon';
+import './index.scss';
+import uploadIcon from '../../data/images/upload.png';
+import infoIcon from '../../data/images/info.png';
+import * as hooks from './hooks';
+import { useDispatch } from 'react-redux';
+
+const messages = defineMessages({
+  dropVideoFileHere: {
+    defaultMessage: 'Drag and drop your videos or',
+    id: 'VideoUploadEditor.dropVideoFileHere',
+  },
+  browse: {
+    defaultMessage: 'Browse files',
+    id: 'VideoUploadEditor.browse',
+  },
+  info: {
+    id: 'VideoUploadEditor.uploadInfo',
+    defaultMessage: 'Upload .mp4 or .mov files (5 GB max)',
+  }
+});
+
+const VideoUploader = ({ onUpload, intl }) => {
+  const [uploadedFiles, setUploadedFile] = useState([]);
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    accept: 'video/*',
+    multiple: true,
+    onDrop: (acceptedFiles) => {
+      setUploadedFile(acceptedFiles);
+      onUpload(acceptedFiles); // Call onUpload prop with the uploaded files
+    },
+  });
+
+  return (
+    <div {...getRootProps({ className: 'dropzone' })} className={`dropzone-wrapper ${isDragActive ? 'active' : ''}`}>
+      <div className="dropzone-middle">
+        <img src={uploadIcon} alt="Upload Icon" />
+        {
+          uploadedFiles.length > 0
+            ? (
+              <ul>
+                {uploadedFiles.map((file) => (
+                  <li key={file.name}>{file.name}</li>
+                ))}
+              </ul>
+            )
+            : (
+              <>
+                <p><FormattedMessage {...messages.dropVideoFileHere}/></p>
+                <Button><FormattedMessage {...messages.browse} /></Button>
+              </>
+            )
+        }
+      </div>
+      <input {...getInputProps()} />
+      <div className="info-line">
+        <img src={infoIcon} alt="Info icon" />
+        <span><FormattedMessage {...messages.info} /></span>
+      </div>
+    </div>
+  );
+};
+
+VideoUploader.propTypes = {
+  onUpload: PropTypes.func.isRequired,
+  intl: intlShape.isRequired,
+};
+
+const VideoUploadEditor = () => {
+  const dispatch = useDispatch();
+
+  const handleDrop = (files) => {
+    const extToMime = {
+      mp4: 'video/mp4',
+      mov: 'video/quicktime',
+    };
+    const supportedFormats = Object.keys(extToMime);
+
+    function getFileExtension(filename) {
+      return filename.slice((filename.lastIndexOf('.') - 1 >>> 0) + 2);
```

Comment:
I'm not entierly sure about the purpose of `>>> 0` here (I could only imagine that it would be used to force a string to be converted to an int, but I don't think it's needed here). And then we do `- 1` and `+ 2`, which results in `+1` which removes the dot before the extension. I think this could be simplified to:
```suggestion
      return filename.slice(filename.lastIndexOf('.') + 1);
```
Or optionally be replaced as:
```suggestion
      return filerename.split('.').pop();
```

Reply:
@Cup0fCoffee This was required because when you drag-and-drop multiple files, the dropzone returns `Blob` objects instead of `File` objects. And so the filenames get some additional characters and doesn't have the `type` parameter.

Reply:
@pkulkark I don't think I understand. I've tried replicating on the devstack by drag and droping multiple files and once as well as doing them one by one, but I'm not seeing anything weird. And even it was, to me it seems like `filename.slice((filename.lastIndexOf('.') - 1 >>> 0) + 2)` and `filename.slice(filename.lastIndexOf('.') + 1)` do the same thing, because (assuming `lastIndexOf` doesn't throw and error because `filename` is `undefined` or `null`), `lastIndexOf` can only return an integer greater or equal to `-1`. So it's `n - 1`, `... >>> 0` doesn't do anything because it's a int already, and then `n - 1 + 2` -> `n + 1`.

Reply:
@Cup0fCoffee Sorry I got confused with another approach I had used before this. The purpose of `>>>0` was just to ensure that the value passed to slice() is never negative. I've updated it to use `Math.abs` to avoid any confusion (and also fix the lint issue about using bitwise operators).